docs/workers.md

```diff
@@ -178,8 +178,11 @@ recommend the use of `systemd` where available: for information on setting up
 
 ### `synapse.app.generic_worker`
 
-This worker can handle API requests matching the following regular
-expressions:
+This worker can handle API requests matching the following regular expressions.
+
+Note that stream writers should have their endpoints routed directly to them for
+efficiency. If this is not done, the requests will be proxied to the proper
+worker.
```

Comment:
this is maybe a simplification? `POST` and `PUT` requests will be proxied, but `GET` requests will be handled directly?

Reply:
I think I missed that detail when looking at the implementation, thanks!